src/java/com/unifina/signalpath/blockchain/WebsocketEthereumJsonRpc.java

```diff
@@ -112,10 +104,17 @@ public boolean openConnectionRetryIfFail() throws InterruptedException {
 			} catch (URISyntaxException e) {
 				log.error(e);
 				return false;
-			} catch (IOException | DeploymentException e) {
+			} catch (IOException | DeploymentException | RuntimeException e) {
 				log.error(e);
 				log.info("Waiting " + attemptReconnectionWaittimeMs + "ms");
-				Thread.sleep(attemptReconnectionWaittimeMs);
+				try {
+					Thread.sleep(attemptReconnectionWaittimeMs);
+				}
+				catch(InterruptedException ie){
+					log.error("Thread was interrupted while sleeping.");
+					//why is this thread interrupted in E&E when sleeping?
```

Comment:
Try `Thread.currentThread().interrupt();` as explained here in detail: https://stackoverflow.com/questions/1087475/when-does-javas-thread-sleep-throw-interruptedexception

Reply:
OK, thanks. Maybe I'll keep throwing the InterruptedException then.

Reply:
Hey this seems a very good point: maybe add the interrupt() after log. Not interrupting the thread might leave the app hanging when it should shut down or something? Should at least try if it works with the `interrupt()`. If it's not broken with it, then please add it. If `interrupt()` breaks it then... maybe just swallow it.

Reply:
i'll test. I don't think the InterruptedException is a major issue in any case

Reply:
Here we should also set `attempts = attemptReconnectionMaxTries` to break out from the loop (or maybe `break`), and the thing will then throw a RuntimeException.

Reply:
If there are multiple threads then InterruptedException should be handled. I don't think that logging exception adds value here but calling interrupt() does.

Reply:
according to the text, the `interrupt()` is kind of a simple fix that shouldn't break anything (if all it does is set a flag), so it should be added; of course if it DOES break things, then add a TODO and think about it later.